src/valuetypes.ts

```diff
@@ -494,40 +494,47 @@ export interface IGuessOptions {
  * @param options additional options
  * @return {any}
  */
-export function guessValueType(editors: ValueTypeEditor[], name: string, index: number, data: any[], accessor: (row: any) => any, options: IGuessOptions = {}): ValueTypeEditor {
+export function guessValueType(editors: ValueTypeEditor[], name: string, index: number, data: any[], accessor: (row: any) => any, options: IGuessOptions = {}): Promise<ValueTypeEditor> {
   options = mixin({
     sampleSize: 100,
     thresholds: <any>{
       numerical: 0.7,
-      categorical: 0.7
+      categorical: 0.7,
+      real: 0.7,
+      int: 0.7
     }
   }, options);
   const testSize = Math.min(options.sampleSize, data.length);
 
-  //compute guess results
-  let results = editors.map((editor) => ({
-    type: editor.id,
-    editor,
-    confidence: editor.isType(name, index, data, accessor, testSize),
-    priority: editor.priority
-  }));
-  //filter all 0 confidence ones by its threshold
-  results = results.filter((r) => typeof options.thresholds[r.type] !== 'undefined' ? r.confidence >= options.thresholds[r.type] : r.confidence > 0);
+  // one promise for each editor for a given column
+  const promises = editors.map((editor) => editor.isType(name, index, data, accessor, testSize));
 
-  if (results.length <= 0) {
-    return null;
-  }
-  //order by priority (less more important)
-  results = results.sort((a, b) => a.priority - b.priority);
-  //choose the first one
-  return results[0].editor;
+  return Promise.all(promises).then((confidenceValues) => {
+    let results = editors.map((editor, i) => ({
+      type: editor.id,
+      editor,
+      confidence: confidenceValues[i],
+      priority: editor.priority
+    }));
+
+    //filter all 0 confidence ones by its threshold
+    results = results.filter((r) => typeof options.thresholds[r.type] !== 'undefined' ? r.confidence >= options.thresholds[r.type] : r.confidence > 0);
+
+    if (results.length <= 0) {
+      return null;
+    }
+    //order by priority (less more important)
+    results = results.sort((a, b) => a.priority - b.priority);
+    //choose the first one
+    return results[0].editor;
+  });
 }
 
-export function createTypeEditor(editors: ValueTypeEditor[], current: ValueTypeEditor, emptyOne = true) {
+export function createTypeEditor(editors: ValueTypeEditor[], current: ValueTypeEditor, emptyOne = true, idType: string = '') {
```

Comment:
the value type of IDType is just one among serveral (string, ...). You shouldn't assume that there are idtypes all the time. 

I would rather go for a combobox that has two levels: 
```
 * string
 * numerical
   * int
   * real --> which removes also the need for the checkbox in the editor
 * idtype
   * ...
```